docs/console/incremental-guide.md

```diff
@@ -0,0 +1,586 @@
+
+If you're unfamiliar with the .NET CLI toolset, read [the overview](overview.md).  It's also helpful to have an understanding of the [console app paradigm](paradigm.md) and how [native binary compilation](single-binaries.md) works.
+
+## Prerequisites
+
+Before you begin, ensure you have the [latest .NET CLI tooling](http://dotnet.github.io/getting-started/).  You'll also need a text editor.
+
+## Hello, Console App!
+
+First, navigate to or create a new folder with a name you like.  "Hello" is the name chosen for the sample code, which can be found [here](https://github.com/dotnet/core-docs/samples/core-projects/console-apps/Hello).
+
+Open up a command prompt and type the following:
+
+```
+$ dotnet new
+$ dotnet restore
+$ dotnet compile
+```
+
+Let's do a quick walkthrough:
+
+1. `$ dotnet init`
+
+   `dotnet init` created an up-to-date `project.json` file with NuGet dependencies necessary to build a console app.  It also created a `Program.cs`, a basic file containing the entry point for the application.
+   
+   `project.json`:
+   ```javascript
+   {
+        "version": "1.0.0-*",
+        "compilationOptions": {
+            "emitEntryPoint": true
+        },
+    
+        "dependencies": {
+            "Microsoft.NETCore.Runtime": "1.0.1-beta-*",
+            "System.IO": "4.0.11-beta-*",
+            "System.Console": "4.0.0-beta-*",
+            "System.Runtime": "4.0.21-beta-*"
+        },
+    
+        "frameworks": {
+            "dnxcore50": { }
+        }
+   }
+   ```
+   `Program.cs`:
+   ```csharp
+   using System;
+
+   namespace ConsoleApplication
+   {
+       public class Program
+       {
+           public static void Main(string[] args)
+           {
+               Console.WriteLine("Hello World!");
+           }
+       }
+   }
+   ```
+
+2. `$ dotnet restore`
+
+   `dotnet restore` analyzed the `project.json` file, downloaded the dependencies stated in the file (or grabbed them from a cache on your machine), and wrote the `project.lock.json` file.  The `project.lock.json` file is necessary to be able to compile and run.
+   
+   The `project.lock.json` file is a persisted and complete set of NuGet dependencies and other information describing an app.  This file is read by other tools, such as `dotnet compile` and `dotnet run`, enabling them to process the source code with a correct set of NuGet dependencies and binding resolutions.
+   
+3. `$ dotnet run`
+
+   `dotnet compile` compiled the source in Intermediate Language (IL) and generated an executable shim and a `Hello.dll` `.dll` file which contains the IL.
+   
+You can now invoke the executable.
+
+```
+$ /bin/Debug/dnxcore50/Hello.exe
+Hello, World!
+```
+
+### Other ways to compile
+
+Let's try compiling `dotnet run` to compile and execute the code without generating any build artifacts.  Note that although this has use (mainly by ASP.NET internally), it's not a recommended way to run build and run console applications.
```

Comment:
@blackdwarf could you provide some context here?  I'm basically trying to explain how `dotnet run` is useful even though it's not part of the regular "flow" of compiling and running a truly native console app.  Explaining that it's a quick way to build+run something like a sample is something I can call out here, but if it has other uses outside of that I think it may be worth mentioning.

Is that command being used (or will it be used) internally by ASP.NET and Visual Studio?